Changelog entry for v0.3.1 release (#1047)
GenericMappingTools/pygmt: doc/changes.md

# Changelog

## Release v0.3.1 (2021/03/14)

[![Digital Object Identifier for PyGMT v0.3.1](https://zenodo.org/badge/DOI/10.5281/zenodo.3781524.svg)](https://doi.org/10.5281/zenodo.3781524)

> *🎉 **Minor changes and bug fixes from v0.3.0** 🎉

Reviewer: I'll let people have a go at choosing the highlights this time :)

```suggestion
### Highlights

*🎉 **Multiple bug fixes and a reorganized gallery** 🎉
*
*
```

* Fix bug that stops Figure.coast from plotting with only dcw parameter ([#910](https://github.com/GenericMappingTools/pygmt/pull/910))

### Documentation

* Add a gallery example showing different line front styles ([#1022](https://github.com/GenericMappingTools/pygmt/pull/1022))
* Add underscore guideline to CONTRIBUTING.md ([#1034](https://github.com/GenericMappingTools/pygmt/pull/1034))
* Add a gallery example for a double y-axes graph ([#1019](https://github.com/GenericMappingTools/pygmt/pull/1019))
* Add a gallery example of inset map showing a rectangle region ([#1020](https://github.com/GenericMappingTools/pygmt/pull/1020))
* Add a gallery example to show coloring of points by categories ([#1006](https://github.com/GenericMappingTools/pygmt/pull/1006))
* Reorganize gallery examples ([#995](https://github.com/GenericMappingTools/pygmt/pull/995))
* Add instructions to upgrade installed PyGMT version ([#1029](https://github.com/GenericMappingTools/pygmt/pull/1029))
* Improve the docstring of the pygmt package ([#1016](https://github.com/GenericMappingTools/pygmt/pull/1016))

> *  Add gallery example showing different polar projection use cases ([#955](https://github.com/GenericMappingTools/pygmt/pull/955))

Reviewer 2: ```suggestion
* Add gallery example showing different polar projection use cases ([#955](https://github.com/GenericMappingTools/pygmt/pull/955))
```

> * Add a gallery example showing different line front styles ([#1022](https://github.com/GenericMappingTools/pygmt/pull/1022))
> * Add underscore guideline to CONTRIBUTING.md ([#1034](https://github.com/GenericMappingTools/pygmt/pull/1034))
> * Add a gallery example for a double y-axes graph ([#1019](https://github.com/GenericMappingTools/pygmt/pull/1019))
> * Add a gallery example of inset map showing a rectangle region ([#1020](https://github.com/GenericMappingTools/pygmt/pull/1020))
> * Add a gallery example to show coloring of points by categories ([#1006](https://github.com/GenericMappingTools/pygmt/pull/1006))
> * Reorganize gallery examples ([#995](https://github.com/GenericMappingTools/pygmt/pull/995))
> * Add instructions to upgrade installed PyGMT version ([#1029](https://github.com/GenericMappingTools/pygmt/pull/1029))
> * Improve the docstring of the pygmt package ([#1016](https://github.com/GenericMappingTools/pygmt/pull/1016))
> * Add gallery example showing different polar projection use cases ([#955](https://github.com/GenericMappingTools/pygmt/pull/955))
> * Add common alias coltypes (-f) for specifying i/o data types ([#994](https://github.com/GenericMappingTools/pygmt/pull/994))
> * Add a gallery example for plotting Cartesian, circular and geographic vectors ([#950](https://github.com/GenericMappingTools/pygmt/pull/950))
> * Expand documentation linking in CONTRIBUTING.md ([#802](https://github.com/GenericMappingTools/pygmt/pull/802))
> * Write changelog in markdown using MyST ([#941](https://github.com/GenericMappingTools/pygmt/pull/941))
> * Update web font to Atkinson Hyperlegible ([#938](https://github.com/GenericMappingTools/pygmt/pull/938))
> * Add a gallery example showing different vector heads and tails ([#890](https://github.com/GenericMappingTools/pygmt/pull/890))
> * Improve the gallery example for datetime inputs ([#919](https://github.com/GenericMappingTools/pygmt/pull/919))

Reviewer 3: Move all entries that "add a gallery example" together:
```suggestion
* Add a gallery example showing different line front styles ([#1022](https://github.com/GenericMappingTools/pygmt/pull/1022))
* Add a gallery example for a double y-axes graph ([#1019](https://github.com/GenericMappingTools/pygmt/pull/1019))
* Add a gallery example of inset map showing a rectangle region ([#1020](https://github.com/GenericMappingTools/pygmt/pull/1020))
* Add a gallery example to show coloring of points by categories ([#1006](https://github.com/GenericMappingTools/pygmt/pull/1006))
* Add gallery example showing different polar projection use cases ([#955](https://github.com/GenericMappingTools/pygmt/pull/955))
* Add a gallery example for plotting Cartesian, circular and geographic vectors ([#950](https://github.com/GenericMappingTools/pygmt/pull/950))
* Add a gallery example showing different vector heads and tails ([#890](https://github.com/GenericMappingTools/pygmt/pull/890))
* Reorganize gallery examples ([#995](https://github.com/GenericMappingTools/pygmt/pull/995))
* Add underscore guideline to CONTRIBUTING.md ([#1034](https://github.com/GenericMappingTools/pygmt/pull/1034))
* Add instructions to upgrade installed PyGMT version ([#1029](https://github.com/GenericMappingTools/pygmt/pull/1029))
* Improve the docstring of the pygmt package ([#1016](https://github.com/GenericMappingTools/pygmt/pull/1016))
* Add common alias coltypes (-f) for specifying i/o data types ([#994](https://github.com/GenericMappingTools/pygmt/pull/994))
* Expand documentation linking in CONTRIBUTING.md ([#802](https://github.com/GenericMappingTools/pygmt/pull/802))
* Write changelog in markdown using MyST ([#941](https://github.com/GenericMappingTools/pygmt/pull/941))
* Update web font to Atkinson Hyperlegible ([#938](https://github.com/GenericMappingTools/pygmt/pull/938))
* Improve the gallery example for datetime inputs ([#919](https://github.com/GenericMappingTools/pygmt/pull/919))
```

* Add a workflow checking links in plaintext and HTML files ([#634](https://github.com/GenericMappingTools/pygmt/pull/634))
* Remove nbsphinx extension ([#931](https://github.com/GenericMappingTools/pygmt/pull/931))
* Improve the error message for loading an old version of the GMT library ([#925](https://github.com/GenericMappingTools/pygmt/pull/925))
* Move requirements-dev.txt dependencies to environment.yml ([#812](https://github.com/GenericMappingTools/pygmt/pull/812))
* Ensure proper non-dev version string when publishing to PyPI ([#900](https://github.com/GenericMappingTools/pygmt/pull/900))
* Run tests in a single CI job (Ubuntu + Python 3.9) for draft PRs ([#906](https://github.com/GenericMappingTools/pygmt/pull/906))

### Contributors

* [Dongdong Tian](https://github.com/seisman)
* [Wei Ji Leong](https://github.com/weiji14)
* [Jiayuan Yao](https://github.com/core-man) 
* [Meghan Jones](https://github.com/meghanrjones)
* [Michael Grund](https://github.com/michaelgrund)
* [Will Schlitzer](https://github.com/willschlitzer)
* [Liam Toney](https://github.com/liamtoney)

> * [Kathryn Materna](https://github.com/kmaterna)

Reviewer 3: There are a few contributors that are not included here, why?

Author: I included everyone in AUTHORS.md; should I have included those who haven't added themselves?

Reviewer: Yes, please include everyone. There should be a git log command to check who commited between v0.3.0 and v0.3.1

Reviewer 3: See here. We want to include every one. https://github.com/GenericMappingTools/pygmt/blob/master/AUTHORSHIP.md#changelog-for-each-release

Author: I added @tawandamoyo and @aliciaha1997, which I think completes the list. Both @michaelgrund and @core-man both look like they had repeat entries when I ran `git shortlog HEAD...v0.3.0 -sne`, and I figured I wouldn't include dependabot (this will definitely be a problem for me if and when there is a robot uprising).

Reviewer: > I figured I wouldn't include dependabot (this will definitely be a problem for me if and when there is a robot uprising).

There should a way to prevent the bot uprising soon with https://github.com/release-drafter/release-drafter/issues/569 :slightly_smiling_face:

# Changelog

## Release v0.3.1 (2021/03/14)

> [![Digital Object Identifier for PyGMT v0.3.1](https://zenodo.org/badge/DOI/10.5281/zenodo.3781524.svg)](https://doi.org/10.5281/zenodo.3781524)

Reviewer: Almost forgot to change the DOI number!

```suggestion
[![Digital Object Identifier for PyGMT v0.3.1](https://zenodo.org/badge/DOI/10.5281/zenodo.4592991.svg)](https://doi.org/10.5281/zenodo.4592991)
```

Author: 🤦🏻‍♂️

### Highlights

> * 🎉 **Multiple bug fixes and a reorganized gallery** 🎉
> * A reorganization of the gallery examples ([#995](https://github.com/GenericMappingTools/pygmt/pull/995))
> * New gallery examples of plotting vectors ([#950](https://github.com/GenericMappingTools/pygmt/pull/950) and [#890](https://github.com/GenericMappingTools/pygmt/pull/890))
> * This is the last PyGMT version to support GMT 6.1.1; the next release will require GMT 6.2

Reviewer: ```suggestion
* 🎉 **Multiple bug fixes and an improved gallery** 🎉
* Reorganized gallery examples into new categories ([#995](https://github.com/GenericMappingTools/pygmt/pull/995))
* Added gallery examples for plotting vectors ([#950](https://github.com/GenericMappingTools/pygmt/pull/950), [#890](https://github.com/GenericMappingTools/pygmt/pull/890))
* Last version to support GMT 6.1.1, future PyGMT versions will require GMT 6.2.0 or newer
```

* Refactor info and grdinfo to use virtualfile_from_data ([#961](https://github.com/GenericMappingTools/pygmt/pull/961))
* Onboarding maintainer checklist ([#773](https://github.com/GenericMappingTools/pygmt/pull/773))
* Add comprehensive tests for pygmt.clib.loading.clib_full_names ([#872](https://github.com/GenericMappingTools/pygmt/pull/872))
* Add a workflow checking links in plaintext and HTML files ([#634](https://github.com/GenericMappingTools/pygmt/pull/634))
* Remove nbsphinx extension ([#931](https://github.com/GenericMappingTools/pygmt/pull/931))
* Improve the error message for loading an old version of the GMT library ([#925](https://github.com/GenericMappingTools/pygmt/pull/925))
* Move requirements-dev.txt dependencies to environment.yml ([#812](https://github.com/GenericMappingTools/pygmt/pull/812))
* Ensure proper non-dev version string when publishing to PyPI ([#900](https://github.com/GenericMappingTools/pygmt/pull/900))
* Run tests in a single CI job (Ubuntu + Python 3.9) for draft PRs ([#906](https://github.com/GenericMappingTools/pygmt/pull/906))

### Contributors

* [Dongdong Tian](https://github.com/seisman)

> * [Wei Ji Leong](https://github.com/weiji14)
> * [Jiayuan Yao](https://github.com/core-man)

Reviewer: @core-man has more commits than me so should move up the spot :)

```suggestion
* [Jiayuan Yao](https://github.com/core-man)
* [Wei Ji Leong](https://github.com/weiji14)
```

> * [Will Schlitzer](https://github.com/willschlitzer)
> * [Meghan Jones](https://github.com/meghanrjones)
> * [Michael Grund](https://github.com/michaelgrund)

Reviewer: All three of you actually have 9 commits (@willschlitzer has 8, +1 after merging this PR), but `git shortlog HEAD...v0.3.0 -sne` sorts things alphabetically so...

```suggestion
* [Meghan Jones](https://github.com/meghanrjones)
* [Michael Grund](https://github.com/michaelgrund)
* [Will Schlitzer](https://github.com/willschlitzer)
```

Author: I realize now why my order was different was that I didn't switch to the master branch; I didn't think I had so many commits!

# Changelog

## Release v0.3.1 (2021/03/14)

[![Digital Object Identifier for PyGMT v0.3.1](https://zenodo.org/badge/DOI/10.5281/zenodo.3781524.svg)](https://doi.org/10.5281/zenodo.3781524)

### Highlights

* 🎉 **Multiple bug fixes and a reorganized gallery** 🎉

> * New gallery example for Cartesian, circular, and geographic vectors ([#950](https://github.com/GenericMappingTools/pygmt/pull/950))

Reviewer: > Then I'd like to recommend another gallery example: [Vector heads and tails](https://www.pygmt.org/dev/gallery/lines/vector_heads_tails.html#sphx-glr-gallery-lines-vector-heads-tails-py). It is cool and useful. How do you think ?@weiji14 @willschlitzer

Yep, actually #950 and #890 are part of the vectors issue at #892 so maybe we could merge it into one bullet point?

For another point, how about mentioning the gallery reorganization #995? There was quite a bit of work (and debate) that went into that!

Reviewer 2: > Yep, actually #950 and #890 are part of the vectors issue at #892 so maybe we could merge it into one bullet point?

Sounds good.

> For another point, how about mentioning the gallery reorganization #995? There was quite a bit of work (and debate) that went into that!

I agree, but the first line says:

> 🎉 **Multiple bug fixes and a reorganized gallery** 🎉

So can we still highlight it again?

Author: Added in 444ef30

Reviewer: > I agree, but the first line says:
> 
> > tada **Multiple bug fixes and a reorganized gallery** tada
> 
> So can we still highlight it again?

Or we can change that first line to say ":tada: Multiple bug fixes and an improved gallery :tada:" :shrug:

Reviewer: Alternatively, we can mention the Twitter account :bird:https://twitter.com/gmt_software

Reviewer 2: > Or we can change that first line to say " Multiple bug fixes and an improved gallery "

That's great. Will update https://hackmd.io/@pygmt/ryt7DI5md when other maintainers approve this PR.

Reviewer: Great, thanks for starting that changelog @core-man! I've added a nice link so using https://hackmd.io/@pygmt/v0-3-1_announcement now:

![image](https://user-images.githubusercontent.com/23487320/111063838-b3116080-8515-11eb-8811-218f1495efc9.png)

Reviewer 4: From my point of view, the most important changes are covered 😉.

# Changelog

## Release v0.3.1 (2021/03/14)

[![Digital Object Identifier for PyGMT v0.3.1](https://zenodo.org/badge/DOI/10.5281/zenodo.3781524.svg)](https://doi.org/10.5281/zenodo.3781524)

### Highlights

> *🎉 **Multiple bug fixes and a reorganized gallery** 🎉

Reviewer: I think we can add one more highlight (usually we have 3 in total). Any more suggestions?

Reviewer 2: Then I'd like to recommend another gallery example: [Vector heads and tails](https://www.pygmt.org/dev/gallery/lines/vector_heads_tails.html#sphx-glr-gallery-lines-vector-heads-tails-py). It is cool and useful.

*

> *

Reviewer: Still need a couple of highlight items here team! What are the key changes to the documentation we want to highlight to our users? Do we want to mention that PyGMT v0.3.1 is the last release to support GMT 6.1.1 (and that the next version will require GMT 6.2.0)?

Reviewer 2: For the documentation, I'd like to recommend [Cartesian, circular, and geographic vectors](https://www.pygmt.org/dev/gallery/lines/vectors.html#sphx-glr-gallery-lines-vectors-py)

> Do we want to mention that PyGMT v0.3.1 is the last release to support GMT 6.1.1 (and that the next version will require GMT 6.2.0)?

Sounds good.

Author: Works for me! Updating now.

> * New gallery examples for Cartesian, circular, and geographic vectors

Reviewer: Add link to the PR here.
```suggestion
* New gallery example for Cartesian, circular, and geographic vectors ([#950](https://github.com/GenericMappingTools/pygmt/pull/950))
```

> * This is the last PyGMT version to support GMT 6.1.1; the next release will require the upcoming GMT 6.2

Reviewer: ```suggestion
* This is the last PyGMT version to support GMT 6.1.1; the next release will require GMT 6.2
```

* Let pygmt.info load datetime columns into a str dtype array ([#960](https://github.com/GenericMappingTools/pygmt/pull/960))
* Check invalid combinations of resolution and registration in load_earth_relief() ([#965](https://github.com/GenericMappingTools/pygmt/pull/965))
* Open figures using the associated application on Windows ([#952](https://github.com/GenericMappingTools/pygmt/pull/952))
* Fix bug that stops Figure.coast from plotting with only dcw parameter ([#910](https://github.com/GenericMappingTools/pygmt/pull/910))

### Documentation

* Add a gallery example showing different line front styles ([#1022](https://github.com/GenericMappingTools/pygmt/pull/1022))
* Add a gallery example for a double y-axes graph ([#1019](https://github.com/GenericMappingTools/pygmt/pull/1019))
* Add a gallery example of inset map showing a rectangle region ([#1020](https://github.com/GenericMappingTools/pygmt/pull/1020))
* Add a gallery example to show coloring of points by categories ([#1006](https://github.com/GenericMappingTools/pygmt/pull/1006))
* Add gallery example showing different polar projection use cases ([#955](https://github.com/GenericMappingTools/pygmt/pull/955))

> * Add a gallery example for plotting Cartesian, circular and geographic vectors ([#950](https://github.com/GenericMappingTools/pygmt/pull/950))

Reviewer: Take this away since it's in the highlights.
```suggestion
```
